pkg/clusteragent/admission/mutate/autoinstrumentation/service_name_mutator.go

```diff
@@ -0,0 +1,120 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoinstrumentation
+
+import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func encodeEnvVar(e corev1.EnvVar) string {
+	out, _ := json.Marshal(e)
+	return string(out)
+}
+
+type serviceEnvVarData struct {
+	env        corev1.EnvVar
```

Comment:
Note to self (and thing to change):

This is still a draft, but I was originally thinking about being _smarter_ here which is where the container pointers come from, and potentially storing whether or not this was an init container or not. 

I'm no longer _also_ iterating through init containers so that's not necessary. 

This function could have potentially output a candidate list of containers missing the env var and we would have a `mutator` that then be executed but it is much simpler to do a different approach, output candidate env vars and pick the best one.